internal/access/grant.go

```diff
@@ -29,30 +30,47 @@ func ListGrants(c *gin.Context, subject uid.PolymorphicID, resource string, priv
 
 	roles := []string{models.InfraAdminRole, models.InfraViewRole, models.InfraConnectorRole}
 	db, err := RequireInfraRole(c, roles...)
-	if err == nil {
-		selectors = append(selectors, data.ByOptionalSubject(subject))
-		return data.ListGrants(db, selectors...)
-	}
 	err = HandleAuthErr(err, "grants", "list", roles...)
-
 	if errors.Is(err, ErrNotAuthorized) {
 		// Allow an authenticated identity to view their own grants
-		db := getDB(c)
-		subjectID, _ := subject.ID()
+		db = getDB(c)
+		subjectID, err2 := subject.ID()
+		if err2 != nil {
+			// user is only allowed to select their own grants, so if the subject is missing or invalid, this is an access error
+			return nil, fmt.Errorf("%w: %s", AuthorizationError{Resource: "grants", Operation: "list", RequiredRoles: []string{"view"}}, err)
```

Comment:
I think this can return `err` directly like it did previously (at the end of the function). That should be the correct authz error:

```suggestion
			// user is only allowed to select their own grants, so if the subject is missing or invalid, this is an access error
			return nil, err
```